File: Fundamentals/Dictionaries - Exercise/softuni_parking.py
count_command = int(input())
softuni_parking = dict()
for i in range(count_command):
    commands = input().split(" ")
    command, username = commands[0], commands[1]
    if command == "register":
        license_plate_number = commands[2]
        if username not in softuni_parking.keys():
            softuni_parking[username] = license_plate_number
            print(f"{username} registered {license_plate_number} successfully")
        else:
            print(f"ERROR: already registered with plate number {license_plate_number}")
    elif command == "unregister":
        if username not in softuni_parking.keys():
            print(F"ERROR: user {username} not found")
        else:
            print(f"{username} unregistered successfully")
            softuni_parking.pop(username)

for username, license_plate_number in softuni_parking.items():
    print(f"{username} => {license_plate_number}")

File: JavaFundamentals/src/MapsLambdaAndStreamAPI/Exercise/SoftUniParking.java
package MapsLambdaAndStreamAPI.Exercise;

import java.util.LinkedHashMap;
import java.util.Map;
import java.util.Scanner;

public class SoftUniParking {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        Map<String, String> parking = new LinkedHashMap<>();

        int n = Integer.parseInt(scanner.nextLine());
        for (int i = 0; i < n; i++) {
            String[] input = scanner.nextLine().split("\\s+");

            String command = input[0];

            switch (command) {
                case "register" -> {
                    String name = input[1];
                    String licensePlateNumber = input[2];

                    if (parking.containsKey(name)) {
                        System.out.printf("ERROR: already registered with plate number %s%n", parking.get(name));
                    } else {
                        parking.put(name, licensePlateNumber);
                        System.out.printf("%s registered %s successfully%n", name, licensePlateNumber);
                    }

                }
                case "unregister" -> {
                    String name = input[1];

                    if (parking.containsKey(name)) {
                        parking.remove(name);
                        System.out.printf("%s unregistered successfully%n", name);
                    } else {
                        System.out.printf("ERROR: user %s not found%n", name);
                    }
                }
            }
        }

        parking.forEach((name, licensePlateNumber) ->
                System.out.printf("%s => %s%n", name, licensePlateNumber));
    }
}
